### PR TITLE
Add component library

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
Add component library backstage including the Section Notice component

#### Ticket Link
NONE

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by M-ZubairAhmed — COMMENTED** [review]
Is this the best place for this to be? I was wondering we were going to do something like storybook? or similar for this one. I believe separation of concerns is important here

**Comment by larkox** [comment]
@M-ZubairAhmed No strong feeling about the location. As long as it is importable from the main product and renderable in the product I am happy with it.

Actually, I have mixed feelings on having the specific component libraries in this folder, or living next to the component they define. 0/5

**Comment by M-ZubairAhmed** [comment]
> @M-ZubairAhmed No strong feeling about the location. As long as it is importable from the main product and renderable in the product I am happy with it.
> 
> Actually, I have mixed feelings on having the specific component libraries in this folder, or living next to the component they define. 0/5

Would be a good topic for guild to discuss

**Review by hmhealey — APPROVED** [review]
A few comments, but this looks like a solid start. I'm excited to play around with it more

**Comment by hmhealey** [comment]
As for where to put this, I feel like I want it more isolated from the rest of the code base, but we don't have a good pattern for doing that yet. We really need something like the mobile app's `screen` folder or something, but I feel like we haven't spent enough time thinking about that

**Comment by larkox** [comment]
I added some "background context" to see the elements in the background they are supposed to be shown. This "fix" the issue I was seeing with theming on dark themes, and surfaces a new issue of the components not looking great when on top of sidebar background (maybe never meant to be on that background, though).

**Review by M-ZubairAhmed — APPROVED** [review]

File: webapp/channels/src/components/component_library/section_notice.cl.tsx [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+const linkButton = {linkButton: {onClick: () => window.alert('link!'), text: 'Link'}};
+
+const SectionNoticeComponentLibrary = () => {
+    const [text, textSelector] = useStringProp('text', 'Some text', true);
+    const [title, titleSelector] = useStringProp('title', 'Some text', false);
+    const [dismissable, dismissableSelector] = useBooleanProp('isDismissable', true);
+    const [sectionType, sectionTypePosibilities, sectionTypeSelector] = useDropdownProp('type', 'danger', sectionTypeValues, true);
+
+    const [showPrimaryButton, setShowPrimaryButton] = useState(false);
+    const onChangePrimaryButton = useCallback((e: React.ChangeEvent<HTMLInputElement>) => setShowPrimaryButton(e.target.checked), []);
```
[the remark this answers is not in the document]
Comment (hmhealey) [reply webapp/channels/src/components/component_library/section_notice.cl.tsx:37]: Ah, I see. I missed that `showPrimaryButton` isn't a prop of the `SectionNotice`. That makes sense to me then

**Comment by larkox** [comment]
@yasserfaraazkhan The only important things to test here is that:

- When the developer settings are enabled, the `component_library` endpoint can be navigated to (`baseURL/component_library`)
- When the developer settings are disabled, the `component_library` endpoint can not be navigated to (it tells you the team cannot be found, or something like that)

You can also play with the tool, and let me know what do you think for testing use :smile:

**Review by yasserfaraazkhan — APPROVED** [review]
Thank you @larkox 

image

**Pull request merged**